backend/courses/views.py

```diff
@@ -376,3 +384,122 @@ def get_queryset(self):
             section__course__semester=get_current_semester(),
             in_add_drop_period=True,
         ).order_by("created_at")
+
+
+# private helper function used to get user's accepted friends
+def get_accepted_friends(user):
+    friendships = Friendship.objects.filter(Q(sender=user) | Q(recipient=user))
+    accepted_friendships = friendships.filter(status=Friendship.FriendshipStatus.ACCEPTED)
+    return [model_to_dict(friendship) for friendship in accepted_friendships]
+
+class FriendshipViewSet(viewsets.ViewSet):
+    model = Friendship
+    queryset = Friendship.objects.all()
```

Comment:
Django has some really nice built-in functionality to auto-generate list methods (e.g. get) from the queryset you specify.
https://docs.djangoproject.com/en/4.1/topics/class-based-views/generic-display/#generic-views-of-objects
We might want to change this to:
```python
queryset = Friendship.objects.none()
def get_queryset(self):
    return Friendship.objects.filter(
        Q(sender=self.request.user) | Q(recipient=self.request.user),
        status__ne=Friendship.Status.REJECTED
    )
```
(including the line above `get_queryset` helps the auto-docs).
Then I think we can just delete the `get` method below. But we would have to make a Friendship [serializer](https://www.django-rest-framework.org/api-guide/serializers/) and include that in the ViewSet with e.g. `serializer_class = FriendshipSerializer`.